interfaces/dbus/backend.go

```diff
@@ -123,10 +159,46 @@ func (b *Backend) Setup(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 	return nil
 }
 
+func (b *Backend) setupServiceActivation(snapInfo *snap.Info, spec interfaces.Specification) error {
+	snapName := snapInfo.InstanceName()
+	s := spec.(*Specification)
+	for _, bus := range []struct {
+		dir      string
+		services map[string]*Service
+	}{
+		{dirs.SnapDBusSessionServicesDir, s.SessionServices()},
+		{dirs.SnapDBusSystemServicesDir, s.SystemServices()},
+	} {
+		globs, content, err := b.deriveContentServices(snapInfo, bus.dir, bus.services)
+		if err != nil {
+			return err
+		}
+		if err := os.MkdirAll(bus.dir, 0755); err != nil {
+			return err
+		}
+		_, _, err = osutil.EnsureDirStateGlobs(bus.dir, globs, content)
+		if err != nil {
+			return fmt.Errorf("cannot synchronize DBus service activation files for snap %q: %s", snapName, err)
+		}
+	}
+	return nil
+}
+
 // Remove removes dbus configuration files of a given snap.
 //
 // This method should be called after removing a snap.
 func (b *Backend) Remove(snapName string) error {
+	if err := b.removeBusConfig(snapName); err != nil {
+		return err
+	}
+	if err := b.removeServiceActivation(snapName); err != nil {
+		return err
+	}
```

Comment:
Do we need to tell DBus daemon that something has changed or is it entirely based on file system notification? Similar question for the case above, where we alter content.

Reply:
There is a `ReloadConfig` method on the bus, but dbus-daemon also performs an inotify watch on its config and service directories and automatically reload.

Debian packaging currently issues a ReloadConfig method call to the system bus when installing/removing services.  It leaves session buses to fend for themselves though.  At this point, it could all be seen as hold over from pre-inotify Linux systems, or to support Debian's non-Linux ports.